### DISCUSSION TITLE
Select All, Undo, Redo not working

### DISCUSSION THREAD

**Discussion by vibsid0986** [eclipse-glsp/glsp#1389] [Q&A]
Hello Team,

I was working on workflow-glsp examples and observed that when we double click on a node and Autocomplete widget opens, select all, undo and redo are not working in there, even their corresponding key bindings seems to select the complete workflow example widget instead of the text in the autocomplete widget. Please see below image when I did Ctrl + A on the node with text "ABC", it selects the complete widget.

Request you to help me in resolving this issue, as I need to implement select all, undo, redo for my custom use case.

Thanks in advance.

![image](https://github.com/user-attachments/assets/b50a89cb-8315-41bd-9db0-d759d99d647c)

**Comment by planger** [comment]
Hi,

Thank you for raising this issue!

You mean that Select All, Undo, Redo, etc are not working in the Autocomplete widget (text field), but are rather handled by the GLSP diagram editor even though the text field had focus? If that's what you mean, this indeed seems like a bug.

In which integration did you observe this bug? Is this VS Code or Eclipse Theia?

I hope this shouldn't be hard to fix. It just needs to ensured that the focused element (text field) catches those events and stops propagation so that they don't bubble up to the diagram parent. Are you planning to provide a fix for it?

This is the underlying code of the Autocomplete widget:
https://github.com/eclipse-glsp/glsp-client/blob/master/packages/client/src/base/auto-complete/auto-complete-widget.ts

Thank you and best wishes,

Philip

**Comment by vibsid0986** [comment]
Hello @planger ,

Thanks for your quick response. I was trying to fix this, however it seems like the event is first triggered on the GLSP editor and then on the auto-complete widget, `event.stopImmediatePropagation()` would have worked if event triggering would have been in reverse order. Also I noticed same issue exists on filter pallete entries option too. Please refer below image for the same.

![image](https://github.com/user-attachments/assets/b8947b6f-fcc5-4e83-8930-6b66226e681b)

Request you to help me out.

Thanks in advance
Siddhant

**Reply by planger** [reply]
Hm, this is surprising... thanks for investigating though! Let's turn this into a bug report and schedule it for triage.
